scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -300,6 +301,25 @@
          (catch Exception e
            (log/error e "Mesos scheduler status update error")))))
 
+(defn write-sandbox-url-to-datomic
+  "Takes a sandbox file server URL from the compute cluster and saves it to datomic."
+  [conn task-id sandbox-url]
+  (try (let [db (db conn)
+             _ (when-not task-id
+                 (throw (ex-info "task-id is nil. Something unexpected has happened."
+                                 {:sandbox-url sandbox-url
+                                  :task-id task-id})))
+             [instance] (first (q '[:find ?i
+                                    :in $ ?task-id
+                                    :where
+                                    [?i :instance/task-id ?task-id]]
+                                  db task-id))]
+         (if (nil? instance)
+           (log/error (str "Sandbox file server URL update error. No instance for task-id " task-id))
+           @(d/transact conn [[:db/add instance :instance/sandbox-url sandbox-url]])))
```

Comment:
Why do we need `:instance/sandbox-url`? Don't we already have `output-url`?

Reply:
we don't. that's computed on the fly

Reply:
Not sure if I understand, can you point me to the code?

Reply:
https://github.com/twosigma/Cook/pull/1293/files#diff-5dd58675d8adc5809600b0ede0bc4713L1013 - `url-path` IS `output-url`
used to be computed from just hostname on the fly

now the new `instance/sandbox-url` is stored in datomic.

Reply:
Why did we switch to storing this in datomic if it can be easily computed from the hostname?

Reply:
Discussed offline with @DaoWen and @nsinkov, the plan is that we're going to store the `container-hostname` in Datomic and then generate the `output-url` from that.